src/api/server.ts

```diff
@@ -0,0 +1,114 @@
+/** Code for managing the connection the RefStudio server. This is only relevant for desktop. */
+
+import { fetch as tauriFetch } from '@tauri-apps/api/http';
+import { Child, ChildProcess, Command as TauriCommand } from '@tauri-apps/api/shell';
+
+export const REFSTUDIO_HOST = 'http://localhost:1487';
+
+interface StartingState {
+  state: 'starting';
+}
+interface ReadyState {
+  state: 'ready';
+  process: Child;
+}
+type ServerState = StartingState | ReadyState;
+
+// A null value means that no one has requested a server yet.
+let serverProcess: ServerState | null = null;
```

Comment:
Lots of ceremony here but I really want to make sure that multiple calls to `getOrStartServer` don't result in multiple servers being started.

Reply:
It's ok.